systemtest/src/test/java/io/strimzi/systemtest/KafkaClusterTest.java

```diff
@@ -253,4 +259,66 @@ public void testClusterWithCustomParameters() {
         assertThat(jsonString, valueOfCmEquals("KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR", "5"));
         assertThat(jsonString, valueOfCmEquals("KAFKA_TRANSACTION_STATE_LOG_REPLICATION_FACTOR", "5"));
     }
+
+    @Test
+    @KafkaCluster(name = "my-cluster", kafkaNodes = 1)
+    public void testDeleteTopicControllerDeployment() {
+        // kafka cluster already deployed via annotation
+        String clusterName = "my-cluster";
+        String topicControllerName = topicControllerDeploymentName(clusterName);
+        LOGGER.info("Running deleteTopicControllerDeployment with cluster {}", clusterName);
+
+        kubeClient.deleteByName(DEPLOYMENT, topicControllerName);
+        kubeClient.waitForResourceDeletion(DEPLOYMENT, topicControllerName);
+
+        LOGGER.info("Waiting for recovery {}", topicControllerName);
+        kubeClient.waitForDeployment(topicControllerName);
+    }
+
+    @Test
+    @KafkaCluster(name = "my-cluster", kafkaNodes = 1)
+    public void testDeleteClusterControllerDeployment() {
+        // kafka cluster already deployed via annotation
+        String clusterName = "my-cluster";
+        String clusterControllerName = "strimzi-cluster-controller";
+        LOGGER.info("Running deleteClusterControllerDeployment with cluster {}", clusterName);
+
+        kubeClient.deleteByName(DEPLOYMENT, clusterControllerName);
+        kubeClient.waitForResourceDeletion(DEPLOYMENT, clusterControllerName);
+
+        LOGGER.info("Waiting for recovery {}", clusterControllerName);
+        kubeClient.waitForDeployment(clusterControllerName);
```

Comment:
As I described in the main comment, the other tests should work thanks to the running cluster controller doing the reconciliation but this test should not work; no-one will recreate the cluster controller if you delete its related Deployment.

Reply:
Yeah, this is valid point - there is nobody to recover the cluster controller. This test doesn't make sense. The other tests do.

Reply:
Absolutely agree, there is no sense to have a test case for deleting CC. I'm going to remove this case